device.go

```diff
@@ -32,6 +32,7 @@ type device struct {
 	BootEnvReadWriter
 	Commander
 	*partitions
+	hasUpdateCb func(d *device) (bool, error)
```

Comment:
It looks to me that this callback is only needed for testability sake, in which case it makes no sense to have it here and instead update the tests to inject proper data as `BootEnvReadWriter` interface.